scikit/notebooks/Scikit Learn on CMLE.ipynb

```diff
@@ -0,0 +1,702 @@
+    "from sklearn.preprocessing import LabelEncoder\n",
+    "encoders = {col:LabelEncoder() for col in CATEGORICAL_COLUMNS}\n",
+    "\n",
+    "for col in CATEGORICAL_COLUMNS:\n",
+    "    train_features[col] = encoders[col].fit_transform(train_features[col])\n",
+    "\n",
+    "for col in CATEGORICAL_COLUMNS:\n",
+    "    test_features[col] = encoders[col].fit_transform(test_features[col])\n",
+    "\n",
+    "\n",
+    "# Create and train a classifier\n",
+    "classifier = RandomForestClassifier()\n",
+    "classifier.fit(train_features, train_labels)\n",
+    "\n",
+    "# Export the classifier to a file\n",
+    "joblib.dump(classifier, 'model.joblib')\n",
+    "\n",
+    "print('Done')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Part 2: Upload the model\n",
+    "Next, you'll need to upload the model to your project's storage bucket in GCS. To use your model with ML Engine, it needs to be uploaded to Google Cloud Storage (GCS). This step takes your local ‘model.joblib’ file and uploads it GCS via the Cloud SDK using gsutil.\n",
+    "\n",
+    "Note: The exact file name of of the exported model you upload to GCS is important! Your model must be named  “model.joblib”, “model.pkl”, or “model.bst” with respect to the library you used to export it. This restriction ensures that the model will be safely reconstructed later by using the same technique for import as was used during export."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 14,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Copying file://./model.joblib [Content-Type=application/octet-stream]...\n",
+      "-\n",
+      "Operation completed over 1 objects/6.5 MiB.                                      \n"
+     ]
+    }
+   ],
+   "source": [
+    "! gsutil cp ./model.joblib gs://$BUCKET_ID/model.joblib"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Part 3: Create a model resource\n",
+    "Cloud ML Engine organizes your trained models using model and version resources. A Cloud ML Engine model is a container for the versions of your machine learning model. For more information on model resources and model versions look [here](https://cloud.google.com/ml-engine/docs/deploying-models#creating_a_model_version). \n",
+    "\n",
+    "At this step, you create a container that you can use to hold several different versions of your actual model."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 12,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Created ml engine model [projects/true-ability-192918/models/census].\r\n"
+     ]
+    }
+   ],
+   "source": [
+    "! gcloud ml-engine models create $MODEL_NAME --regions us-central1"
```

Comment:
How does CMLE know its scikit model? There is no model type needed to be specified?

Reply:
this is specified in version resource. users can technically have v1 be sklearn and v2 be tf